jsx/StaticDataTable.js

```diff
@@ -340,6 +356,11 @@ var StaticDataTable = React.createClass({
     var filteredRows = this.countFilteredRows();
     var currentPageRow = (rowsPerPage * (this.state.PageNumber - 1));
     var filteredData = [];
+    var useKeyword = false;
+
+    if (this.props.Filter.keyword) {
+      useKeyword = true;
+    }
```

Comment:
i'm confused as to why lines 361-363 is already in `countFilteredRows()` function on line 356, then `useKeyword` is set to false again, and then back to true again. Perhaps a couple comments would help

Reply:
@zaliqarosli The search keyword functionality was already a feature of the the un-reactified version of the module. Due to how the StaticDataTable works, the functionality of the searchKeyword was broken when the module was reactified; hence, I made sure in this PR that none of the existing functionality of Data Dictionary is lost when it is reactified. 

Regarding this.props.Filter.keyword - in all the modules where there is keyword search functionality, the actual PHP element being added is always called `keyword`. 

The existing function `countFilteredRows()` is responsible for displaying the # of filtered rows being displayed out of all the rows that have been filtered for. In order to ensure that there isn't a disparity in those two numbers, the same check has to be made in `hasFilterKeyword()` and `countFilteredRows`.

Reply:
ahh gotcha! (regarding the first two points)

i don't see where hasFilterKeyword() comes into play here. it looks to me like the lines i was commenting on is in the render function?

Reply:
@zaliqarosli oops - I meant to say that the check has to be done in the render function and countFilteredRows since the render function is where the rows to be displayed are determined.